### ISSUE TITLE
^ anchor does not work

### ISSUE THREAD

**Issue by smackesey** [phaazon/hop.nvim#147]
```lua
require('hop').hint_patterns({}, '\\v^.')
```

This should highlight the first character in every line,  instead it appears to highlight every character in the file. Interestingly, the corresponding `\\v.$` *does* work.

(NOTE: I know about `HopLine`, I'm trying to use the above pattern as one part of a longer regex).

**Comment by shyun3** [comment]
I think this is because [hint_patterns()](https://github.com/phaazon/hop.nvim/blob/7fb2504044dfb83ddfa4cfd42bf56f1861409864/lua/hop/init.lua#L274) uses [jump_target.regex_by_case_searching()](https://github.com/phaazon/hop.nvim/blob/7fb2504044dfb83ddfa4cfd42bf56f1861409864/lua/hop/init.lua#L299) which sets [oneshot = false](https://github.com/phaazon/hop.nvim/blob/7fb2504044dfb83ddfa4cfd42bf56f1861409864/lua/hop/jump_target.lua#L342). I'm not an expert in the codebase but this sounds like the pattern search will be repeated on the remainder of the string that wasn't matched. If there is a string remaining, there will always be a beginning so effectively all characters will be matched. For the `$` example, there is no remaining string after the end of a string so this search works correctly.

The simplest, hackiest approach might be to scan the pattern for a `^` at the beginning. If this is found, then `oneshot` can be set to true, otherwise false.

**Comment by phaazon** [comment]
Yep, that doesn’t work. I’m not sure exactly why. `oneshot` is to be used for matcher that will match only once per-line, so even if you find `^` only once, repeating the regex for the rest of the line with `oneshot = false` will just be a waste of resource; it should still return the right answer.

**Comment by Zantier** [comment]
It definitely seems to be matching `^` with the end of the previous match. I was hoping I could use negative lookbehind as a workaround, by replacing all `^` with `.@<!` (assuming `\v`), but the same issue happens.
